modules/tensor_mechanics/src/materials/ADComputeFiniteStrainElasticStress.C

```diff
@@ -16,15 +16,26 @@ ADComputeFiniteStrainElasticStress::validParams()
 {
   InputParameters params = ADComputeStressBase::validParams();
   params.addClassDescription("Compute stress using elasticity for finite strains");
+  params.addParam<bool>(
+      "use_old_elasticity_tensor",
+      false,
+      "Flag to optionally use the elasticity tensor computed at the previous timestep.");
   return params;
 }
 
 ADComputeFiniteStrainElasticStress::ADComputeFiniteStrainElasticStress(
     const InputParameters & parameters)
   : ADComputeStressBase(parameters),
     GuaranteeConsumer(this),
+    _use_old_elasticity_tensor(getParam<bool>("use_old_elasticity_tensor")),
     _elasticity_tensor_name(_base_name + "elasticity_tensor"),
-    _elasticity_tensor(getADMaterialProperty<RankFourTensor>(_elasticity_tensor_name)),
+    _elasticity_tensor(
+        _use_old_elasticity_tensor
+            ? getGenericZeroMaterialProperty<RankFourTensor, true>(_elasticity_tensor_name)
+            : getGenericMaterialProperty<RankFourTensor, true>(_elasticity_tensor_name)),
+    _elasticity_tensor_old(_use_old_elasticity_tensor
+                               ? getMaterialPropertyOld<RankFourTensor>(_elasticity_tensor_name)
+                               : getZeroMaterialProperty<RankFourTensor>("zero")),
```

Comment:
```suggestion
                               : getZeroMaterialProperty<RankFourTensor>()),
```

This change is a must and it depends on #16000.